src/PufferDepositorV2.sol

```diff
@@ -55,9 +55,11 @@ contract PufferDepositorV2 is IPufferDepositorV2, PufferDepositorStorage, Access
         }) { } catch { }
 
         SafeERC20.safeTransferFrom(IERC20(address(_WST_ETH)), msg.sender, address(this), permitData.amount);
-        uint256 stETHAmount = _WST_ETH.unwrap(permitData.amount);
 
-        return PUFFER_VAULT.depositStETH(stETHAmount, recipient);
+        _WST_ETH.unwrap(permitData.amount);
+
+        // The PufferDepositor is not supposed to hold any stETH, so we sharesOf(PufferDepositor) to the PufferVault immediately
+        return PUFFER_VAULT.depositStETH(_ST_ETH.sharesOf(address(this)), recipient);
```

Comment:
Let's have a test scenario for this as well (for when stETHAmount < shareOf(address(this)) ), just to make sure it doesn't mess up any accountings on the vault.